node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -196,6 +196,8 @@ case class DataMessageHandler(
           logger.error(s"Error when processing headers message", err)
         }
 
+        callbacks.onBlockHeadersReceived.foreach(_.apply(headers))
```

Comment:
This line should be changed to something like:
```suggestion
for {
  _ <- callbacks.onBlockHeadersReceived.foldLeft(FutureUtil.unit)(
              (acc, callback) => acc.flatMap(_ => callback(headers)))
  newApi <- chainApiF
...
```